jobsdb/jobsdb_test.go

```diff
@@ -36,251 +33,196 @@ import (
+			require.True(t, jd.lifecycle.started)
 		})
 
-		It("can call Start twice without side-effects", func() {
-			Expect(jd.Start()).To(BeNil())
+		t.Run("can call Start twice without side-effects", func(t *testing.T) {
+			jd := startTestJobsDB(t)
+			defer jd.TearDown()
+			require.NoError(t, jd.Start())
 			group1 := jd.backgroundGroup
-			Expect(jd.Start()).To(BeNil())
+			require.NoError(t, jd.Start())
 			group2 := jd.backgroundGroup
-			Expect(group1).To(Equal(group2))
-			Expect(jd.lifecycle.started).To(Equal(true))
+			require.Equal(t, group1, group2)
+			require.True(t, jd.lifecycle.started)
 		})
 
-		It("can call Start in parallel without side-effects", func() {
+		t.Run("can call Start in parallel without side-effects", func(t *testing.T) {
+			jd := startTestJobsDB(t)
+			defer jd.TearDown()
 			var wg sync.WaitGroup
 			bgGroups := make([]*errgroup.Group, 10)
 			wg.Add(10)
 			for i := 0; i < 10; i++ {
 				idx := i
 				go func() {
-					Expect(jd.Start()).To(BeNil())
+					require.NoError(t, jd.Start())
 					bgGroups[idx] = jd.backgroundGroup
 					wg.Done()
 				}()
 			}
 			wg.Wait()
 			for i := 1; i < 10; i++ {
-				Expect(bgGroups[i-1]).To(Equal(bgGroups[i]))
+				require.Equal(t, bgGroups[i-1], bgGroups[i])
 			}
-			Expect(jd.lifecycle.started).To(Equal(true))
+			require.True(t, jd.lifecycle.started)
 		})
 
-		It("can call Stop twice without side-effects", func() {
-			Expect(jd.Start()).To(BeNil())
-			Expect(jd.lifecycle.started).To(Equal(true))
-			Expect(jd.backgroundGroup).ToNot(BeNil())
+		t.Run("can call Stop twice without side-effects", func(t *testing.T) {
+			jd := startTestJobsDB(t)
+			defer jd.TearDown()
+			require.NoError(t, jd.Start())
+			require.True(t, jd.lifecycle.started)
+			require.NotNil(t, jd.backgroundGroup)
 			jd.Stop()
-			Expect(jd.backgroundGroup).ToNot(BeNil())
-			Expect(jd.lifecycle.started).To(Equal(false))
-			Expect(jd.backgroundGroup.Wait()).To(BeNil())
+			require.NotNil(t, jd.backgroundGroup)
+			require.False(t, jd.lifecycle.started)
+			require.NoError(t, jd.backgroundGroup.Wait())
 			jd.Stop()
-			Expect(jd.backgroundGroup).ToNot(BeNil())
-			Expect(jd.lifecycle.started).To(Equal(false))
-			Expect(jd.backgroundGroup.Wait()).To(BeNil())
+			require.NotNil(t, jd.backgroundGroup)
+			require.False(t, jd.lifecycle.started)
+			require.NoError(t, jd.backgroundGroup.Wait())
 		})
 
-		It("can call Stop in parallel without side-effects", func() {
-			Expect(jd.Start()).To(BeNil())
+		t.Run("can call Stop in parallel without side-effects", func(t *testing.T) {
+			jd := startTestJobsDB(t)
+			defer jd.TearDown()
+			require.NoError(t, jd.Start())
 
 			var wg sync.WaitGroup
 			wg.Add(10)
 			for i := 0; i < 10; i++ {
 				go func() {
 					jd.Stop()
-					Expect(jd.backgroundGroup.Wait()).To(BeNil())
+					require.NoError(t, jd.backgroundGroup.Wait())
 					wg.Done()
 				}()
 			}
 			wg.Wait()
-		})
-
-		It("can call Start & Stop in parallel without problems", func() {
```

Comment:
Don't we need this test?

Reply:
we do yeah.
missed it while cleaning up, my bad.

Reply:
added it back